csrc/device_lower/pass/replace_size.cpp

```diff
@@ -35,131 +37,68 @@ namespace {
 // concice there to pull out. May want to consider making this mapping its own
 // class especially as it may be useful during scheduling.
 std::unordered_map<Val*, Val*> getSimplificationMap(Fusion* fusion) {
-  std::list<std::unordered_set<IterDomain*>> disjoint_root_sets;
-  std::unordered_map<IterDomain*, std::unordered_set<IterDomain*>*>
-      id_to_disjoint_root_set;
-
-  auto map_root_ids = [&disjoint_root_sets, &id_to_disjoint_root_set](
-                          IterDomain* id0, IterDomain* id1) {
-    if (id0->isBroadcast() || id1->isBroadcast()) {
-      return;
-    }
-
-    if (id0->isGatherScatter() || id1->isGatherScatter()) {
-      return;
-    }
```

Comment:
Wondering if there was a reason for special-casing for these cases. Instead I am just using `getMaybeExpandedExtent()` now.